src/post/workshed.ts

```diff
@@ -81,13 +81,14 @@ const worksheds = [
       }
       if (!get("trainsetConfiguration")) {
         print("Reconfiguring trainset, as our next station is empty", "blue");
-        return TrainSet.setConfiguration(getBestCycle());
+        return TrainSet.setConfiguration(getRotatedCycle(get("trainsetPosition") % 8));
       } else {
-        const bestTwoStations = getBestCycle().splice(0, 2);
-        const offset = get("trainsetPosition") % 8;
-        if (bestTwoStations.includes(TrainSet.next())) return;
+        const bestStations = getPrioritizedStations();
+        if (bestStations.includes(TrainSet.next())) return;
+        if (globalOptions.ascend && estimatedTurns() <= 40) return;
         print(`Reconfiguring trainset, as our next station is ${TrainSet.next()}`, "blue");
-        return TrainSet.setConfiguration(offsetDefaultPieces(offset));
+        const position = get("trainsetPosition") % 8;
+        return TrainSet.setConfiguration(getRotatedCycle(position));
```

Comment:
Minor style nitpicks:
`get("trainsetPosition") % 8` is assigned to `position` in the `else` case, but not in the `if` case.
Also `if (globalOptions.ascend && estimatedTurns() <= 40) return;` can probably be pushed before the `getPrioritizedStations()` compute, since it is cheaper